### DISCUSSION TITLE
Cant write QSPI partitions on the Orin NX.

### DISCUSSION THREAD

**Discussion by isaacporras** [OE4T/meta-tegra#1423] [Q&A]
Hi, when attempting to flash the Orin NX, I'm getting the following error:

```
I> MB2 (version: 0.0.0.0-t234-54845784-9c429857)
I> t234-A01-0-Silicon (0x12347)
I> Boot-mode : RCM Flash
I> Emulation:
I> Entry timestamp: 0x01e8bf9f
I> Regular heap: [base:0x40040000, size:0x10000]
I> DMA heap: [base:0x269800000, size:0x800000]
I> Task: ARI update carveout TZDRAM (0x50001e80)
I> Task: PSC mailbox init (0x50016a7c)
I> Task: Map CCPLEX_INTERWORLD_SHMEM carveout (0x50001e2c)
I> Task: Program CBB PCIE AMAP regions (0x50019b64)
I> Task: Boot device init (0x50001d74)
I> Boot_device: QSPI_FLASH instance: 0
I> Qspi clock source : pllc_out0
I> QSPI-0l initialized successfully
I> Task: Partition Manager Init (0x50001d70)
W> Cannot find any partition table for 00030000
 > PARTITION_MANAGER: Failed to publish partition.
I> Task: Trigger mailbox for PSC-BL1 exit (0x50016afc)
I> Sending opcode 0x4d420802 to psc
I> Received ACK from psc
I> Task: Enter 3p server (0x50001f18)
I> USB configuration success
I> Populate storage info
I> Erasing device 3: 0
I> QSPI: Erasing entire device
I> Writing device 3: 0.
W> Cannot find any partition table for 00030000
E> NV3P_SERVER: Failed to initialize partition table from GPT.
```

The serial output is:

```
[   7.5639 ] Start flashing
[   7.5745 ] tegradevflash_v2 --pt flash.xml.bin --create
[   7.5765 ] Bootloader version 01.00.0000
[   7.6066 ] Erasing spi: 0 ......... [Done]
[   8.6160 ] Writing partition secondary_gpt with gpt_secondary_3_0.bin [ 16896 bytes ]
[   8.6175 ] [................................................] 100%
[   8.6214 ] 000000004d4d2c01: E> NV3P_SERVER: Failed to initialize partition table from GPT.
[   8.8449 ]
[   8.8450 ]
Error: Return value 1
Command tegradevflash_v2 --pt flash.xml.bin --create
```

The flashing command that I'm using is:

`sudo ./doflash.sh`

I

**Comment by dwalkes** [comment]
Hi @isaacporras are you using https://github.com/OE4T/tegra-demo-distro ?  Which image(s) and branch(es) have you tried?

**Reply by isaacporras** [reply]
Hi Dan, 
No, I'm not using the `tegra-demo-distro`. I'm currently flashing my custom carrier board (with Orin NX 8GB SOM) with a custom build with machine based on `p3509-a02-p3767-0000`.

**Comment by dwalkes** [comment]
> attempting to flash the Orin NX,
> The flashing command that I'm using is: `sudo ./doflash.sh`

I think you'll need to use https://github.com/OE4T/meta-tegra/wiki/initrd-flashing-support with Orin NX, since I believe that's only going to support NVMe external storage.

**Reply by isaacporras** [reply]
I'm currently using doflash script as suggested by Matt M [here](https://github.com/OE4T/meta-tegra/discussions/1318) due to a previous issue.

**Reply by dwalkes** [reply]
Thanks for the cross link.  I think what Matt was suggesting there was `sudo ./doflash.sh --spi-only` followed by `sudo ./initrd-flash --skip-bootloader`

**Reply by isaacporras** [reply]
Yes, however the flashing process fails at the first command `sudo ./doflash.sh --spi-only`

**Reply by dwalkes** [reply]
Got it, so above when you wrote
```
The flashing command that I'm using is:

sudo ./doflash.sh
```
You meant
```
The flashing command that I'm using is:

sudo ./doflash.sh --spi-only
```
and the logs are from a spi-only run?  If so I think the thing to look at is why the secondary GPT write step would fail in spi flash and the cause of these prints:
```
I> Task: Partition Manager Init (0x50001d70)
W> Cannot find any partition table for 00030000
 > PARTITION_MANAGER: Failed to publish partition.
```

Have you tried flashing with stock L4T and comparing the `flash.sh` script behavior to what the scripts are doing and the contents of flash.xml? This might be where I'd start, that and [mining the developer forum](https://forums.developer.nvidia.com/search?expanded=true&q=%20Cannot%20find%20any%20partition%20table%20for) for similar error cases to see if this message is related.

**Comment by paroque28** [comment]
@isaacporras I'm facing a similar issue.
Were you able to flash this board? What were the main changes?
I appreciate everything you can share

**Reply by dchvs** [reply]
@paroque28, you can run: `sudo ./doflash.sh --no-flash --rcm-boot --external-device`,  in order to solve this issue. Hope it helps.
